tests/framework/extensions/users/users.go

```diff
@@ -0,0 +1,69 @@
+package users
+
+import (
+	"time"
+
+	"github.com/rancher/norman/types"
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	management "github.com/rancher/rancher/tests/framework/clients/rancher/generated/management/v3"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+func CreateUserWithRole(rancherClient *rancher.Client, user *management.User, role string) (*management.User, error) {
+	createdUser, err := rancherClient.Management.User.Create(user)
+	if err != nil {
+		return nil, err
+	}
+
+	roleBinding := &management.GlobalRoleBinding{
+		GlobalRoleID: role,
+		UserID:       createdUser.ID,
+	}
+
+	_, err = rancherClient.Management.GlobalRoleBinding.Create(roleBinding)
+	if err != nil {
+		return nil, err
+	}
+
+	return createdUser, nil
+}
+
+func AddProjectMember(rancherClient *rancher.Client, project *management.Project, user *management.User, projectRole string) error {
+	role := &management.ProjectRoleTemplateBinding{
+		ProjectID:       project.ID,
+		UserPrincipalID: user.PrincipalIDs[0],
+		RoleTemplateID:  projectRole,
+	}
+
+	err := wait.Poll(1*time.Second, 1*time.Minute, func() (bool, error) {
```

Comment:
we can assume the project exists here we need to wait for the condition `BackingNamespaceCreated`. We should also use a watch.

Reply:
fixed